### PR TITLE
chore: test clean up and refactor

### PR DESCRIPTION
**Pull request by Shahroz16**
PR helps cleanup tests.

What is being removed/updated 
- Tests that aren't used anymore
- HTTP test has been updated for common code and HTTP client test has been moved to Push messaging modules
- Queue test has been removed and code that is only being used for test now, is now in test directory rather than source.

### CODE REVIEW

File: Sources/Common/Background Queue/QueueStorage.swift [hunk 1]
```diff
@@ -34,13 +34,13 @@ public protocol QueueStorage: AutoMockable {
 public class FileManagerQueueStorage: QueueStorage {
     private let fileStorage: FileStorage
     private let jsonAdapter: JsonAdapter
-    private let siteId: String
+    let siteId: String
```
Comment (Shahroz16) [inline Sources/Common/Background Queue/QueueStorage.swift:37]: made public for testing as this is just used for testing and migration

File: Tests/MessagingPush/HttpClientTest.swift [hunk 2]
```diff
@@ -0,0 +1,179 @@
+@testable import CioInternalCommon
+@testable import CioMessagingPush
+import Foundation
+@testable import SharedTests
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+import XCTest
+
+class HttpClientTest: UnitTest {
```
Comment (Shahroz16) [inline Tests/MessagingPush/HttpClientTest.swift:10]: moved the client here as HTTP client that its testing is MessagingPush specific

File: Tests/Tracking/migration/QueueExtension.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
+@testable import CioInternalCommon
+import Foundation
+
+public enum QueueTaskGroup {
```
Comment (Shahroz16) [inline Tests/Tracking/migration/QueueExtension.swift:4]: removed it from common and added it in test as its only used in testing

File: Tests/Tracking/CustomerIOIntegrationTests.swift [hunk 4]
```diff
@@ -1,166 +0,0 @@
-@testable import CioInternalCommon
-@testable import CioTracking
-import Foundation
-import SharedTests
-import XCTest
-
-class CustomerIOIntegrationTests: IntegrationTest {
```
Comment (mrehan27) [inline Tests/Tracking/CustomerIOIntegrationTests.swift:7]: Why is this removed? 🤔

File: Tests/MessagingPush/HttpTest.swift [hunk 5]
```diff
@@ -18,24 +19,23 @@ import XCTest
...
-        if let siteId = getEnvironmentVariable("SITE_ID"), let apiKey = getEnvironmentVariable("API_KEY") {
+        if let writeKey = getEnvironmentVariable("WRITE_KEY") {
             runner = UrlRequestHttpRequestRunner()
-            session = CIOHttpClient.getCIOApiSession(
-                siteId: siteId,
-                apiKey: apiKey,
-                userAgentHeaderValue: userAgentUtil.getUserAgentHeaderValue()
+            session = RichPushHttpClient.getCIOApiSession(
+                key: writeKey,
+                userAgentHeaderValue: deviceInfo.getUserAgentHeaderValue()
```
Comment (mrehan27) [inline Tests/MessagingPush/HttpTest.swift:38]: Aren't we using `userAgentUtil` anymore? Because looking at the code, `userAgentHeaderValue: userAgentUtil.getUserAgentHeaderValue()` made more sense to me than `userAgentHeaderValue: deviceInfo.getUserAgentHeaderValue()` 😅

**Review by mrehan27 — APPROVED** [review]
Just a few comments, looks good me otherwise 👍🏻

File: Tests/MessagingPush/HttpTest.swift (hunk 5 above)
Comment (Shahroz16) [reply Tests/MessagingPush/HttpTest.swift:38]: We still have user agent class, but what the useragent class was doing was combining device info information with the client information. But since we don't have any client information in the NSE. We just need device info for this purpose
